### PR TITLE
Support both rv32/rv64 on qemu-rv

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
* Support rv32/rv64 on single virt board
* Let ISA configuable to make evaluate NuttX on risc-v easily
## Impact
qemu-rv32 only
## Testing
CI and QEMU

### CODE REVIEW

File: arch/risc-v/Kconfig [hunk 1]
```diff
@@ -113,8 +113,6 @@ config ARCH_CHIP_RV32M1
 config ARCH_CHIP_QEMU_RV32
 	bool "QEMU RV32"
 	select ARCH_RV32
-	select ARCH_RV_ISA_M
-	select ARCH_RV_ISA_A
```
Comment (xiaoxiang781216) [inline arch/risc-v/Kconfig:117]: why limit the ISA set if qemu always support them?

Comment (no1wudi) [reply arch/risc-v/Kconfig:117]: Now these options moved to `qemu-rv`'s Kconfig to provide more flexiblity.
```
config ARCH_CHIP_QEMU_RV_ISA_M
    bool "Standard Extension for Integer Multiplication and Division"
    default y
    select ARCH_RV_ISA_M

config ARCH_CHIP_QEMU_RV_ISA_A
    bool "Standard Extension for Atomic Instructions"
    default y
    select ARCH_RV_ISA_A

config ARCH_CHIP_QEMU_RV_ISA_C
    bool "Standard Extension for Compressed Instructions"
    default y
    select ARCH_RV_ISA_C
```

Comment (xiaoxiang781216) [reply arch/risc-v/Kconfig:117]: If so why not make ARCH_RV_ISA_F/ARCH_RV_ISA_D configurable?

File: boards/risc-v/qemu-rv/rv-virt/configs/nsh/defconfig [hunk 2]
```diff
@@ -18,18 +18,22 @@
...
-CONFIG_ARCH_BOARD_QEMU_RV32_VIRT=y
-CONFIG_ARCH_CHIP="qemu-rv32"
+CONFIG_ARCH_BOARD="rv-virt"
+CONFIG_ARCH_BOARD_QEMU_RV_VIRT=y
+CONFIG_ARCH_CHIP="qemu-rv"
 CONFIG_ARCH_CHIP_QEMU_RV32=y
+CONFIG_ARCH_CHIP_QEMU_RV=y
+CONFIG_ARCH_CHIP_QEMU_RV_ISA_A=y
+CONFIG_ARCH_CHIP_QEMU_RV_ISA_C=y
+CONFIG_ARCH_CHIP_QEMU_RV_ISA_M=y
```
[the remark this answers is not in the document]
Comment (no1wudi) [reply boards/risc-v/qemu-rv/rv-virt/configs/nsh/defconfig:36]: Yes, but these items are generated by savedefconfig, and need them to pass the CI.

File: arch/risc-v/Kconfig (hunk 1 above)
Comment (no1wudi) [reply arch/risc-v/Kconfig:117]: They are controlled by CONFIG_ARCH_FPU and CONFIG_ARCH_DPFPU eventually.

Comment (xiaoxiang781216) [reply arch/risc-v/Kconfig:117]: ARCH_RV_ISA_F/ARCH_RV_ISA_D is same as ARCH_HAVE_FPU/SRCH_HAVE_DFPU, should we remove the first?

File: boards/risc-v/qemu-rv/rv-virt/configs/nsh/defconfig (hunk 2 above)
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply boards/risc-v/qemu-rv/rv-virt/configs/nsh/defconfig:36]: if so, why not change the default from y ro n?

File: arch/risc-v/Kconfig (hunk 1 above)
Comment (no1wudi) [reply arch/risc-v/Kconfig:117]: Done

File: boards/risc-v/qemu-rv/rv-virt/configs/nsh/defconfig (hunk 2 above)
[the remark this answers is not in the document]
Comment (no1wudi) [reply boards/risc-v/qemu-rv/rv-virt/configs/nsh/defconfig:36]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**